node_modules/oae-authentication/tests/test-auth.js

```diff
@@ -0,0 +1,107 @@
+         */
+        it('verify requesting login ids for a user', function(callback) {
+            // Generate a user id
+            var userId = TestsUtil.generateTestUserId();
+
+            // Verify that an error is thrown when a malformed user id was specified
+            RestAPI.Authentication.getUserLoginIds(globalAdminRestContext, userId, function(err, loginIds) {
+                assert.ok(err);
+                assert.equal(err.code, 400);
+
+                // Verify that an error is thrown when an unexisting user id was specified
+                RestAPI.Authentication.getUserLoginIds(globalAdminRestContext, 'u:camtest:abcdefghij', function(err, loginIds) {
+                    assert.ok(err);
+                    assert.equal(err.code, 404);
+
+                    // Create a test user
+                    RestAPI.User.createUser(camAdminRestContext, userId, 'password', 'Test User', null, function(err, createdUser) {
+                        assert.ok(!err);
+
+                        // Verify that an error is thrown when an anonymous user on the global admin router requests the login ids for the test user
+                        RestAPI.Authentication.getUserLoginIds(anonymousGlobalRestContext, createdUser.id, function(err, loginIds) {
+                            assert.ok(err);
+                            assert.equal(err.code, 401);
+
+                            // Verify that an error is thrown when an anonymous user on the tenant router requests the login ids for the test user
+                            RestAPI.Authentication.getUserLoginIds(anonymousCamRestContext, createdUser.id, function(err, loginIds) {
+                                assert.ok(err);
+                                assert.equal(err.code, 401);
+
+                                // Verify that an error is thrown when an unauthorized tenant admin requests the login ids for the test user
+                                RestAPI.Authentication.getUserLoginIds(gtAdminRestContext, createdUser.id, function(err, loginIds) {
+                                    assert.ok(err);
+                                    assert.equal(err.code, 401);
+
+                                    // Verify that a tenant admin can request the login ids for the test user
+                                    RestAPI.Authentication.getUserLoginIds(camAdminRestContext, createdUser.id, function(err, loginIds) {
+                                        assert.ok(!err);
+                                        assert.ok(loginIds);
+
+                                        // Verify that a global admin can request the login ids for the test user
+                                        RestAPI.Authentication.getUserLoginIds(globalAdminRestContext, createdUser.id, function(err, loginIds) {
+                                            assert.ok(!err);
+                                            assert.ok(loginIds);
+
+                                            callback();
```

Comment:
The unit tests should include tests that verify the login ids we expect are returned in the response. To avoid having a single test that gets way too big, it would probably be worth having one `it` block for validation+authorization tests, and then one block for ensuring the validity of the login ids.